webrender/src/render_backend.rs

```diff
@@ -514,4 +528,134 @@ impl RenderBackend {
         let mut notifier = self.notifier.lock();
         notifier.as_mut().unwrap().as_mut().unwrap().new_scroll_frame_ready(composite_needed);
     }
+
+
+    #[cfg(not(feature = "debugger"))]
+    fn get_docs_for_debugger(&self) -> String {
+        String::new()
+    }
+
+    // TODO(gw): The display list traversal below is a copy of the skeleton
+    //           from frame.rs. We might like to try and unify the traversal
+    //           functions so that this code could be shared, and invoke
+    //           a callback as items are encountered?
```

Comment:
Or maybe implement Display or Debug or some specific method on display items such that you could do something like:

```rust
match *item.item() {
    SpecificDisplayItem::PushStackingContext(..) => {
        let mut subtraversal = item.sub_iter();
        let mut child_node = debug_server::TreeNode::new("stacking context");
        self.traverse_items(&mut subtraversal, &mut child_node);
        node.add_child(child_node);
        return Some(subtraversal);
    }
    other_item => { node.add_item(format!("{}", other_item)); }
}
```

Reply:
Ah, good idea - that's much tidier.